inference-engine/src/mkldnn_plugin/cpu_blocked_memory_desc.cpp

```diff
@@ -0,0 +1,246 @@
+    const MemoryDesc* pRhs = &rhs;
+    if (auto blockingDesc = dynamic_cast<const BlockedMemoryDesc*>(pRhs)) {
+        return isCompatible(*blockingDesc);
+    } else if (auto mkldnnDesc = dynamic_cast<const MKLDNNMemoryDesc*>(pRhs)) {
+        return mkldnnDesc->isCompatible(*this);
+    } else {
+        return false;
+    }
+}
+
+bool BlockedMemoryDesc::isCompatible(const BlockedMemoryDesc& rhs) const {
+    if (this->getShape() != rhs.getShape() || this->getPrecision() != rhs.getPrecision())
+        return false;
+
+    if (!isEqualOrUndefined(this->getBlockDims(), rhs.getBlockDims())) {
+        return false;
+    }
+
+    if (!isEqualOrUndefined(this->getOffsetPaddingToData(), rhs.getOffsetPaddingToData())) {
+        return false;
+    }
+
+    size_t skipAxis = this->getShape().getRank() > 0 && this->getShape().getDims().front() == 1 ? 0 :
+            Shape::UNDEFINED_DIM; //ignore batch axis if batch size == 1
+    if (!isEqualOrUndefined(this->getStrides(), rhs.getStrides(), skipAxis)) {
+        return false;
+    }
+
+    if (!isEqualOrUndefined(this->getOrder(), rhs.getOrder())) {
+        return false;
+    }
+
+    return dimsEqualWeak(this->getOffsetPadding(), rhs.getOffsetPadding());
+}
+
+size_t BlockedMemoryDesc::getMemSizeImp() const {
+    int64_t e_size = getOffsetPadding() + 1;  // size in bytes (from begin of data to last element)
+    for (int j = 0; j < getBlockDims().size(); j++)
+        e_size += (getBlockDims()[j] - 1) * getStrides()[j];
+
+    // In some cases computational formula above doesn't work properly (e.g. for OhIw8o4i layout).
+    // This WA allows to limit the size of allocated memory from below.
+    // TODO: need to properly investigate the root cause of incorrect computations
+    int64_t min_size = 1;
+    for (int64_t dim : getBlockDims()) {
+        min_size *= dim;
+    }
+    e_size = std::max(e_size, min_size);
+
+    e_size *= getPrecision() == InferenceEngine::Precision::BIN ? 1 : getPrecision().size();
+
+    return e_size;
+}
+
+size_t BlockedMemoryDesc::getOffset(const InferenceEngine::SizeVector& v) const {
+    InferenceEngine::SizeVector off_v = v;
+
+    size_t n_blocked_dims = order.size();
+    if (blockedDims.size() != n_blocked_dims || strides.size() != n_blocked_dims) {
+        IE_THROW() << "Cannot calculate offset. Incorrect primitive descriptor!";
+    }
+    InferenceEngine::SizeVector blockedShift(n_blocked_dims);
+    for (size_t i = 1; i <= n_blocked_dims; i++) {
+        blockedShift[n_blocked_dims - i] = off_v[order[n_blocked_dims - i]] % blockedDims[n_blocked_dims - i];
+        off_v[order[n_blocked_dims - i]] /= blockedDims[n_blocked_dims - i];
+    }
+    size_t offset = getOffsetPadding();
+    for (size_t d = 0; d < n_blocked_dims; ++d) {
+        const size_t p = blockedShift[d] + getOffsetPaddingToData()[d];
+        offset += p * strides[d];
+    }
+    return offset;
+}
+
+size_t BlockedMemoryDesc::getOffset(size_t elemNumber) const {
```

Comment:
I agree that it is slow. But in not performance critical contexts, such as blob dumping, this method gives really convenient way to read the blob without additional complexity connected with its internal representation. M.b. it would be better to keep this method in the public section, but provide some warning message that this method is slow and must not be used in performance critical contexts?

Reply:
Rename to `getElementOffset` and move to the private section.

Reply:
```MKLDNNSplitNode``` also used ```getElementOffset``` inside ```optimizedNspc2Ncsp```, so ```MKLDNNSplitNode``` added as friend into ```MemoryDesc```

Reply:
It is better to add specific method the friend list that use this call instead of the entire class.

Reply:
```optimizedNspc2Ncsp``` is private method ```MKLDNNSplitNode```